src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1970,7 +1983,7 @@ void SetTotalDifficultyDeep(BlockHeader current)
                 }
             }
 
-            if (header.TotalDifficulty is not null)
+            if (header.TotalDifficulty is not null && header.TotalDifficulty != 0)
```

Comment:
This is checked in multiple places, maybe encapsulate in a method or somehow else. Use UInt256.Zero to avoid casting.